addons/account/models/account_bank_statement.py

```diff
@@ -253,11 +253,11 @@ def write(self, values):
             # by marking those record as needing to be recomputed.
             # Note that marking the field is not enough as we also have to recompute all its other fields that are depending on 'previous_statement_id'
             # hence the need to call modified afterwards.
-            to_recompute = self.search([('previous_statement_id', 'in', self.ids), ('id', 'not in', self.ids)])
+            to_recompute = self.search([('previous_statement_id', 'in', self.ids), ('id', 'not in', self.ids), ('journal_id', '=', self.journal_id.id)])
             if to_recompute:
                 self.env.add_to_compute(self._fields['previous_statement_id'], to_recompute)
                 to_recompute.modified(['previous_statement_id'])
-            next_statements_to_recompute = self.search([('previous_statement_id', 'in', [st.previous_statement_id.id for st in self]), ('id', 'not in', self.ids)])
+            next_statements_to_recompute = self.search([('previous_statement_id', 'in', [st.previous_statement_id.id for st in self]), ('id', 'not in', self.ids), ('journal_id', '=', self.journal_id.id)])
```

Comment:
I'd guess that it could make sense to add `('previous_statement_id', '!=', False)`. There should not be 2 statements in a journal with `previous_statement_id` set to `False`.

However, the condition on the journal could be sufficient for our issue.